nixos/modules/system/etc/build-composefs-dump.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+
+"""Build a composefs dump from a Json config"""
+
+import functools
+import glob
+import json
+import os
+import sys
+from enum import Enum
+from typing import Any
+
+Attrs = dict[str, Any]
+
+
+class FileType(Enum):
+    directory = "4"
+    file = "10"
```

Comment:
I've added a link to the man page of composefs-dump which describes the format of the resulting dump. I don't think its super valuable to document the inner workings of this script for now since I expect to have quite a bit of movement here. If you want to understand it, there is no way around reading the script for now.